### DISCUSSION TITLE
v2.x series and new DB Engine?

### DISCUSSION THREAD

**Discussion by Tra5is** [codenotary/immudb#1878] [General]
The 1.9 release has the message:

> immudb version 1.9DOM marks the final release in the 1.x series. We are working on the new v2.x series with a whole new db engine

I'm curious to know more details to help determine if I should invest time and effort integrating with the current 1.9 implementation. 

How will the new engine be different?
Will the exiting clients remain compatible with the existing gRPC API?
What might the upgrade look like? (Just update the immudb binary? Or will the DB need to be migrated? Is the new engine optional?)
What is the motivation behind a new DB Engine? Should we expect performance, security, functional improvements?

**Comment by vchaindz** [comment]
It's too early to disclose details, but you can continue building apps on 1.9. It won't be a breaking change.
